File: client/src/services/IProductService.ts
import { ProductModel } from "@/models/ProductModel";

export interface IProductService {
    getAll(pageNum: Number, pageSize: Number): Promise<ProductModel>;
    getById(id: Number): Promise<ProductModel>;
    getByCategory(categoryId: Number, pageNum: Number, pageSize: Number): Promise<ProductModel[]>;
    getProductImagePath(id: Number): string;
}

File: server/pbl03api/src/main/java/com/benkyousuru/pbl03api/model/service/IProductService.java
package com.benkyousuru.pbl03api.model.service;

import java.io.IOException;
import java.util.List;

import com.benkyousuru.pbl03api.model.model.ProductModel;

public interface IProductService extends ICrudService<ProductModel, Integer> {
    List<ProductModel> getAll(int pageNumber, int pageSize);
    List<ProductModel> getByCategory(int categoryId, int pageNumber, int pageSize);
    byte[] getImageById(int productId) throws IOException;
}
